local exports = {}

exports[`json > should serialize the machine 1`] = [[
{
  "entry": [],
  "exit": [],
  "history": false,
  "id": "(machine)",
  "initial": {
    "actions": [],
    "eventType": null,
    "source": "#(machine)",
    "target": [
      "#(machine).foo",
    ],
  },
  "invoke": [
    {
      "id": "invokeId",
      "src": "invokeSrc",
      "type": "xstate.invoke",
    },
  ],
  "key": "(machine)",
  "on": {},
  "order": -1,
  "output": {
    "result": 42,
  },
  "states": {
    "bar": {
      "entry": [],
      "exit": [],
      "history": false,
      "id": "(machine).bar",
      "initial": {
        "actions": [],
        "eventType": null,
        "source": "#(machine).bar",
        "target": [],
      },
      "invoke": [],
      "key": "bar",
      "on": {},
      "order": 3,
      "states": {},
      "tags": [],
      "transitions": [],
      "type": "atomic",
      "version": "1.0.0",
    },
    "foo": {
      "entry": [],
      "exit": [],
      "history": false,
      "id": "(machine).foo",
      "initial": {
        "actions": [],
        "eventType": null,
        "source": "#(machine).foo",
        "target": [],
      },
      "invoke": [],
      "key": "foo",
      "on": {},
      "order": 2,
      "states": {},
      "tags": [],
      "transitions": [],
      "type": "atomic",
      "version": "1.0.0",
    },
    "testActions": {
      "entry": [
        {
          "type": "stringActionType",
        },
        {
          "type": "objectActionType",
        },
        {
          "other": "any",
          "type": "objectActionTypeWithExec",
        },
        {
          "type": "actionFunction",
        },
        {
          "type": "xstate.assign",
        },
        {
          "type": "xstate.assign",
        },
        {
          "type": "xstate.raise",
        },
      ],
      "exit": [
        {
          "type": "xstate.cancel",
        },
      ],
      "history": false,
      "id": "(machine).testActions",
      "initial": {
        "actions": [],
        "eventType": null,
        "source": "#(machine).testActions",
        "target": [],
      },
      "invoke": [
        {
          "id": "invokeId",
          "src": "invokeSrc",
          "type": "xstate.invoke",
        },
      ],
      "key": "testActions",
      "on": {
        "TO_FOO": [
          {
            "actions": [],
            "eventType": "TO_FOO",
            "reenter": false,
            "source": "#(machine).testActions",
            "target": [
              "#(machine).foo",
              "#(machine).bar",
            ],
          },
        ],
        "xstate.after.1000.(machine).testActions": [
          {
            "actions": [],
            "delay": 1000,
            "event": "xstate.after.1000.(machine).testActions",
            "eventType": "xstate.after.1000.(machine).testActions",
            "reenter": false,
            "source": "#(machine).testActions",
            "target": [
              "#(machine).bar",
            ],
          },
        ],
      },
      "order": 1,
      "states": {},
      "tags": [],
      "transitions": [
        {
          "actions": [],
          "eventType": "TO_FOO",
          "reenter": false,
          "source": "#(machine).testActions",
          "target": [
            "#(machine).foo",
            "#(machine).bar",
          ],
        },
        {
          "actions": [],
          "delay": 1000,
          "event": "xstate.after.1000.(machine).testActions",
          "eventType": "xstate.after.1000.(machine).testActions",
          "reenter": false,
          "source": "#(machine).testActions",
          "target": [
            "#(machine).bar",
          ],
        },
      ],
      "type": "atomic",
      "version": "1.0.0",
    },
    "testFinal": {
      "entry": [],
      "exit": [],
      "history": false,
      "id": "(machine).testFinal",
      "initial": {
        "actions": [],
        "eventType": null,
        "source": "#(machine).testFinal",
        "target": [],
      },
      "invoke": [],
      "key": "testFinal",
      "on": {},
      "order": 5,
      "output": {
        "something": "else",
      },
      "states": {},
      "tags": [],
      "transitions": [],
      "type": "final",
      "version": "1.0.0",
    },
    "testHistory": {
      "entry": [],
      "exit": [],
      "history": "deep",
      "id": "(machine).testHistory",
      "initial": {
        "actions": [],
        "eventType": null,
        "source": "#(machine).testHistory",
        "target": [],
      },
      "invoke": [],
      "key": "testHistory",
      "on": {},
      "order": 4,
      "states": {},
      "tags": [],
      "transitions": [],
      "type": "history",
      "version": "1.0.0",
    },
    "testParallel": {
      "entry": [],
      "exit": [],
      "history": false,
      "id": "(machine).testParallel",
      "initial": {
        "actions": [],
        "eventType": null,
        "source": "#(machine).testParallel",
        "target": [],
      },
      "invoke": [],
      "key": "testParallel",
      "on": {},
      "order": 6,
      "states": {
        "one": {
          "entry": [],
          "exit": [],
          "history": false,
          "id": "(machine).testParallel.one",
          "initial": {
            "actions": [],
            "eventType": null,
            "source": "#(machine).testParallel.one",
            "target": [
              "#(machine).testParallel.one.inactive",
            ],
          },
          "invoke": [],
          "key": "one",
          "on": {},
          "order": 7,
          "states": {
            "inactive": {
              "entry": [],
              "exit": [],
              "history": false,
              "id": "(machine).testParallel.one.inactive",
              "initial": {
                "actions": [],
                "eventType": null,
                "source": "#(machine).testParallel.one.inactive",
                "target": [],
              },
              "invoke": [],
              "key": "inactive",
              "on": {},
              "order": 8,
              "states": {},
              "tags": [],
              "transitions": [],
              "type": "atomic",
              "version": "1.0.0",
            },
          },
          "tags": [],
          "transitions": [],
          "type": "compound",
          "version": "1.0.0",
        },
        "two": {
          "entry": [],
          "exit": [],
          "history": false,
          "id": "(machine).testParallel.two",
          "initial": {
            "actions": [],
            "eventType": null,
            "source": "#(machine).testParallel.two",
            "target": [
              "#(machine).testParallel.two.inactive",
            ],
          },
          "invoke": [],
          "key": "two",
          "on": {},
          "order": 9,
          "states": {
            "inactive": {
              "entry": [],
              "exit": [],
              "history": false,
              "id": "(machine).testParallel.two.inactive",
              "initial": {
                "actions": [],
                "eventType": null,
                "source": "#(machine).testParallel.two.inactive",
                "target": [],
              },
              "invoke": [],
              "key": "inactive",
              "on": {},
              "order": 10,
              "states": {},
              "tags": [],
              "transitions": [],
              "type": "atomic",
              "version": "1.0.0",
            },
          },
          "tags": [],
          "transitions": [],
          "type": "compound",
          "version": "1.0.0",
        },
      },
      "tags": [],
      "transitions": [],
      "type": "parallel",
      "version": "1.0.0",
    },
  },
  "tags": [],
  "transitions": [],
  "type": "compound",
  "version": "1.0.0",
}
]];

return exports
